services/movie_session.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+from django.db.models import QuerySet
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+        show_time=movie_show_time
+    )
+
+
+def get_movies_sessions(session_date: Optional[str] = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: Optional[datetime] = None,
+        movie_id: Optional[int] = None,
+        cinema_hall_id: Optional[int] = None
+) -> Optional[MovieSession]:
+    if not any([show_time, movie_id, cinema_hall_id]):
+        return None
+
+    movie_session = get_movie_session_by_id(session_id)
+    if movie_session is None:
+        return None
+
+    update_fields = {}
```

Comment:
You can just leave lines 45-57. No need for any extra checks, no need to return queryset in this function